tests/publiccloud/check_registercloudguest.pm

```diff
@@ -44,7 +44,23 @@ sub run {
         return;
     }
 
-    if (is_container_host()) {
+    # this is special use case needed by dev ops team within SUSE. Should not be used for any product testing.
+    # Use this variable ONLY if you really know what you are doing!
+    if (check_var('PUBLIC_CLOUD_INFRA', '1')){
+        my $prefix = "PREPARATION PHASE.";
+        my $rmt_ipv4 = get_required_var("PUBLIC_CLOUD_INFRA_RMT_V4");
+        my $rmt_ipv6 = get_required_var("PUBLIC_CLOUD_INFRA_RMT_V6");
+        if(is_ondemand()) {
+            check_instance_unregistered($instance, "$prefix BYOS instance was registered before we started");
+            $instance->ssh_assert_script_run(cmd => "sudo ${path}registercloudguest --clean", fail_message => "$prefix registercloudguest --clean failed");
+            $instance->ssh_assert_script_run(cmd => 'sudo rm /var/log/cloudregister; sudo rm /etc/pki/trust/anchors/*.pem', fail_message => "$prefix Deletion of register log and/or certificates failed");
```

Comment:
Do you really need to remove the crt anchors?

Reply:
ok , make sense

Reply:
I am just following instructions from infra team . it is their take :man_shrugging:

Reply:
and so you do `assert_script_run(cmd => 'ls /');` as well? No, you just do `assert_script_run('ls /');`.